src/render.py

```diff
@@ -1,83 +1,124 @@
+
 
 def print_board_place_marble(board):
-    print("\n        A  B  C   D  E  F")
-    print("     ┌──────────+─────────┐")
+    print("\n       A  B  C   D  E  F")
+    print("     ┌─────────+─────────┐")
 
     for x, line in enumerate(board, 0):
         if x != 0 and x % HALF_BOARD_SIZE == 0:
-            print("     |──────────+─────────|")
-        
-        print("  " + str(x + 1) + generate_line_values(line))
+            print("     |─────────+─────────|")
+
+        print("  " + str(x + 1) + generate_line_values(x, line))
+
+    print("     └─────────+─────────┘\n\n")
 
-    print("     └──────────+─────────┘\n")
 
 def print_board_rotate(board):
     print("\n      2 ↻             3 ↺")
-    print(" 1 ↺ ┌──────────+─────────┐  4 ↻")
+    print(" 1 ↺ ┌─────────+─────────┐  4 ↻")
 
     for x, line in enumerate(board, 0):
         if x != 0 and x % HALF_BOARD_SIZE == 0:
-            print("     |──────────+─────────|")
+            print("     |─────────+─────────|")
 
-        print("   " + generate_line_values(line))
+        print("   " + generate_line_values(x, line))
 
-    print(" 8 ↻ └──────────+─────────┘ 5 ↺")
+    print(" 8 ↻ └─────────+─────────┘ 5 ↺")
     print("      7 ↺             6 ↻ \n")
 
+
+def print_board_finished(board, correct_combinations):
+    print("\n     The game is finished")
+    print("     ┌─────────+─────────┐")
+
+    for x, line in enumerate(board, 0):
+        if x != 0 and x % HALF_BOARD_SIZE == 0:
+            print("     |─────────+─────────|")
+
+        print("   " + generate_line_values(x, line, correct_combinations))
+
+    print("     └─────────+─────────┘")
+    print("\n")
+
+
 """
 board = 2d array
 mode = PRINT_BOARD_PLACE_MARBLE : display X & Y coordonates on the header and on the left side.
 mode = PRINT_BOARD_ROTATE : display rotation input to indicate which enter to rotate each quarters
 """
-def print_board(board, mode=PRINT_BOARD_PLACE_MARBLE):
+
+
+def print_board(board, mode, correct_combinations):
 
     if mode == PRINT_BOARD_PLACE_MARBLE:
         print_board_place_marble(board)
-    else:
+    elif mode == PRINT_BOARD_ROTATE:
         print_board_rotate(board)
+    else:
+        print_board_finished(board, correct_combinations)
+
 
 def print_player(player, current_player_id):
     if player.id == current_player_id:
-        print("    | " + get_marble_character(player.id) + " " + player.name + " |", end="")
+        print("    | " + get_player_color(player.id, PLAYER_CHARACTER) +
+              " " + player.name + " |", end="")
     else:
-        print("    " + get_marble_character(player.id) + " " + player.name, end="")
+        print("     " + get_player_color(player.id, PLAYER_CHARACTER) + "  " + player.name, end="")
 
 
 def print_players(players, current_player_id):
 
     # Constant used to display box around player 1's nickname or player 2.
 
-    BOX_SPACES = " " * 14 * (current_player_id - 1)
+    BOX_SPACES = " " * 16 * (current_player_id - 1)
```

Comment:
magic number, what 16 means ?

Reply:
`len("Player 1") = 8` + 4 spaces to the left + 4 between 2 players.

It's not the best but I working actually on a refacto in the next PR for drawing this box because players will have nicknames.
Also I have added a constant `BASE_SPACES=4` to be more rigorous.